lib/elixir/test/elixir/kernel/fancy_diagnostics_test.exs

```diff
@@ -74,6 +74,32 @@ defmodule Kernel.FancyDiagnosticsTest do
       assert strip_ansi(output) == expected
     end
 
+    test "trims leading spaces on snippet" do
+      expected = """
+      ** (SyntaxError) invalid syntax found on nofile:5:11:
+         ┌─ error: nofile:5:11
+         │
+       5 │ end
+         │ ^
+         │
+         unexpected reserved word: end
+      """
+
+      output =
+        capture_raise(
+          ~S("""
+          if true do 
+              """
+                
+          end
+          """),
+          SyntaxError
+        )
+
+      assert ansi_error?(output)
+      assert strip_ansi(output) == expected
```

Comment:
It is not guaranteed the OS we are running on supports ansi. So asserting on ansi will fail unless we explicitly enable ansi escape esquences. But, if we are going this far, maybe we should disable ansi escapes when running the tests? I am not worried about testing that because testing for the presence of the ansi doesn't mean much, it can still look ugly. :)

Reply:
I don't think we need to disable it, we can just strip it if any and stop ensuring there are ansi codes.
The test samples get very noise if we keep adding ansi codes to them, so better to check it as plain text.

Reply:
So let's disable ansi by writing to the application environment. We already marked the whole test as async anyway. :)